src/libs/highlevel/elektra.c

```diff
@@ -0,0 +1,473 @@
+}
+
+kdb_octet_t elektraGetOctetArrayElement (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_OCTET (getArrayElementValueAsString (elektra, name, KDB_TYPE_OCTET, index));
+}
+
+kdb_short_t elektraGetShortArrayElement (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_SHORT (getArrayElementValueAsString (elektra, name, KDB_TYPE_SHORT, index));
+}
+
+kdb_unsigned_short_t elektraGetUnsignedShortArrayElement  (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_UNSIGNED_SHORT (getArrayElementValueAsString (elektra, name, KDB_TYPE_UNSIGNED_SHORT, index));
+}
+
+kdb_long_t elektraGetLongArrayElement (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_LONG (getArrayElementValueAsString (elektra, name, KDB_TYPE_LONG, index));
+}
+
+kdb_unsigned_long_t elektraGetUnsignedLongArrayElement (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_UNSIGNED_LONG (getArrayElementValueAsString (elektra, name, KDB_TYPE_UNSIGNED_LONG, index));
+}
+
+kdb_long_long_t elektraGetLongLongArrayElement (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_LONG_LONG (getArrayElementValueAsString (elektra, name, KDB_TYPE_LONG_LONG, index));
+}
+
+kdb_unsigned_long_long_t elektraGetUnsignedLongLongArrayElement (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_UNSIGNED_LONG_LONG (getArrayElementValueAsString (elektra, name, KDB_TYPE_UNSIGNED_LONG_LONG, index));
+}
+
+kdb_float_t elektraGetFloatArrayElement (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_FLOAT (getArrayElementValueAsString (elektra, name, KDB_TYPE_FLOAT, index));
+}
+
+kdb_double_t elektraGetDoubleArrayElement (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_DOUBLE (getArrayElementValueAsString (elektra, name, KDB_TYPE_DOUBLE, index));
+}
+
+kdb_long_double_t elektraGetLongDoubleArrayElement (Elektra * elektra, const char * name, size_t index)
+{
+    return KDB_STRING_TO_LONG_DOUBLE (getArrayElementValueAsString (elektra, name, KDB_TYPE_LONG_DOUBLE, index));
+}
+
+// Private functions
+
+void setValueAsString (Elektra * elektra, const char * name, const char * value, KDBType type)
+{
+    int ret = 0;
+    do
+    {
+        Key * const lookupKey = generateLookupKey (elektra, name);
+        Key * const key = lookup (elektra, lookupKey);
+        checkType (key, type);
+
+        keySetString (key, value);
+
+        ret = kdbSet (elektra->kdb, elektra->config, elektra->parentKey);
+        if (ret == -1)
+        {
+            Key * problemKey = ksCurrent (elektra->config);
+            if (problemKey != NULL)
+            {
+//                ELEKTRA_LOG_DEBUG("problemKey: %s\n", keyName (problemKey));
+            }
+
+            kdbGet (elektra->kdb, elektra->config, elektra->parentKey);
```

Comment:
Unnecessary kdbGet if kdbSet was successful.

Please take a look at examples/kdbset.c or qt-gui for how to do this correctly. You also need to do a 3-way merge considering who changed the value (for the one key to set).